### DISCUSSION TITLE
Which languages are you mostly familiar with to develop Wasm apps?

### DISCUSSION THREAD

**Discussion by alabulei1** [WasmEdge/WasmEdge#1443] [Polls]
WebAssembly could support 20+ languages. Which languages are you mostly familiar with to develop Wasm apps?

**Comment by alabulei1** [comment]
Updates:

1. From Reddit, developers also use **WAT and AssemblyScrip**t to develop Wasm apps. 

2. Sorry for my mistake. It should be JavaScript. I made a typo here.

**Reply by ColinEberhardt** [reply]
By JavaScript do you actually mean JavaScript running via a JS-Engine compiled to wasm? Or do you mean a Js-derived language like AssemblyScript?

**Reply by juntao** [reply]
We use QuickJS compiled into Wasm. And then we customize QuickJS to add WasmEdge extensions as JS APIs -- such as the http networking module, Tensorflow module etc. 

https://wasmedge.org/book/en/dev/js/quickstart.html

**Reply by ColinEberhardt** [reply]
Yeah, my point is, I most often use AssemblyScript to develop wasm apps. But I would not choose JavaScript in that poll as they are not the same 😉

**Reply by juntao** [reply]
Right. Definitely not the same. However, from the developer's point of view, there may be little difference -- say if we offer a function as a service based on Wasm, developers can upload their functions in Rust, AssemblyScript, and JavaScript. 😅

It is like all those "JavaScript compilers" -- they are not really compilers either.
